pkgs/by-name/jo/joplin-desktop/package.nix

```
@@ -1,26 +1,187 @@
 {
   lib,
   stdenv,
-  appimageTools,
-  fetchurl,
+  nodejs,
+  makeDesktopItem,
+  copyDesktopItems,
   makeWrapper,
-  _7zz,
+  fetchFromGitHub,
+  yarn-berry_3,
+  python3,
+  pkg-config,
+  pango,
+  cairo,
+  pixman,
+  libsecret,
+  electron,
+  xcbuild,
+  buildPackages,
+  callPackage,
```

I don't think there's any value in having 2 different packages that just return the same app built differently.

If we have a source-code-based packaging option, then we don't need the binary one at all, right?

Even in the case you want to keep it for some time as a fallback in case the new packaging has problems, I'd just add a new argument to the closure like this:

```suggestion
  callPackage,
  packagingMode ? "binary" if stdenv.hostPlatform.isDarwin else "source",
```

Then, inside this same folder, you can have a `package-from-source.nix` and a `package-from-bin.nix`.

---

Thank you for the review. I agree that there is no value in keeping the -bin package once the source built one can fully replace it (i.e. not break Darwin). Unless someone raises another reason to keep it, I'd remove it once the mentioned PR is merged and the other issue fixed.

Maybe putting both packages under `pkgs/by-name/jo/joplin-desktop` like you suggest would be a good idea, yes. I think I'd prefer that to having `joplin-desktop-source`, `joplin-desktop-bin` and an alias `joplin-desktop`, when it's only supposed to be temporary after all.

---

made the change

---

Another thought on this: packages that use `fetchYarnBerryDeps` apparently land on cache.nixos.org for Darwin just fine. So only building it locally, with sandbox enabled, won't work (only if the offlineCache would change even, otherwise e.g. overriding the package should be completely fine too). I think that doesn't actually need to be a blocker for replacing the binary package completely.

The other problem I mentioned is probably still relevant until the fix lands in master: https://nixpk.gs/pr-tracker.html?pr=416077
(Unless hydra builds darwin packages with sandbox disabled)